prov/efa/src/rxr/rxr.h

```diff
@@ -776,13 +776,26 @@ struct rxr_rx_entry *rxr_ep_alloc_rx_entry(struct rxr_ep *ep,
 static inline void rxr_release_rx_entry(struct rxr_ep *ep,
 					struct rxr_rx_entry *rx_entry)
 {
+	struct rxr_pkt_entry *pkt_entry;
+	struct dlist_entry *tmp;
+
 	if (rx_entry->peer)
 		ofi_atomic_dec32(&rx_entry->peer->use_cnt);
 
 #if ENABLE_DEBUG
 	dlist_remove(&rx_entry->rx_entry_entry);
 #endif
-	assert(dlist_empty(&rx_entry->queued_pkts));
+	if (!dlist_empty(&rx_entry->queued_pkts)) {
```

Comment:
Could you explain which test triggered these two bugs? 
Also I'm not quite follow why we need to release queued pkt_entry and rx_entry here. 
e.g., for `rx_entry->queued_entry`, isn't we remove it in progress engine, if we successfully post the ctrl pkt or send out all queued pkt?

Reply:
memory poisoning triggered this bug. With memory poisoning, the rx_entry->queued_entry is detroyed, so when progress engine try to remote it, it hit segfault.

Reply:
>  why we need to release queued pkt_entry and rx_entry->queued_entry here.

queued_pkts and queued_entry are all resources associated with this rx_entry, so we should release them we we release rx_entry. This became particularly important when we remove a staled peer.

Reply:
Now I see, the iallgather test.